### PR TITLE
Add POST /metadata route for adjusting metadata on an existing file

### PR DESCRIPTION
**Pull request by acron0**
Add ability to update metadata

### CODE REVIEW

File: src/witan/httpapi/components/activities.clj [hunk 1]
```diff
@@ -75,18 +76,26 @@
       (nil? receipt)                                      [404 nil nil]
       (not= (:kixi.user/id receipt) (:kixi.user/id user)) [401 nil nil]
       (= "pending" (::spec/status receipt))               [202 nil nil]
-      :else [303 nil {"Location" (::spec/uri receipt)}])))
+      (and (= "complete" (::spec/status receipt))
+           (nil? (::spec/uri receipt)))                   [200 nil nil]
+      (= "complete" (::spec/status receipt))              [303 nil {"Location" (::spec/uri receipt)}])))
```
Comment (otfrom) [inline src/witan/httpapi/components/activities.clj]: I'm torn about having constants like `(def SEE_OTHER 303)` just so we don't have to look up what the more obscure 3xx and 2xx codes are.

File: src/witan/httpapi/components/activities.clj [hunk 2]
```diff
@@ -227,31 +231,34 @@
...
   nil)
 
+(defmethod on-event
+  [:kixi.datastore.file-metadata/updated "1.0.0"]
+  [db {:keys [kixi.comms.event/payload] :as event}]
+  (when (= (::kdcs/file-metadata-update-type payload) ::kdcs/file-metadata-update)
+    (let [command-id (:kixi.comms.command/id event)]
+      (when-let [receipt (retreive-receipt db command-id)]
+        (let [file-id (:kixi.datastore.metadatastore/id payload)]
+          (complete-receipt! db command-id nil))))))
```
Comment (otfrom) [inline src/witan/httpapi/components/activities.clj]: `file-id` isn't used. why is this last let here?

File: src/witan/httpapi/components/activities.clj [hunk 3]
```diff
@@ -227,31 +231,34 @@
         (complete-receipt! db command-id (str "/api/files/" file-id "/metadata")))))
   nil)
 
+(defmethod on-event
+  [:kixi.datastore.file-metadata/updated "1.0.0"]
+  [db {:keys [kixi.comms.event/payload] :as event}]
+  (when (= (::kdcs/file-metadata-update-type payload) ::kdcs/file-metadata-update)
+    (let [command-id (:kixi.comms.command/id event)]
+      (when-let [receipt (retreive-receipt db command-id)]
```
Comment (otfrom) [inline src/witan/httpapi/components/activities.clj]: `receipt` isn't used. This doesn't need to be a when-let

File: src/witan/httpapi/spec.clj [hunk 4]
```diff
@@ -31,128 +39,75 @@
...
-(s/def ::file-metadata
-  (s/keys :req [:kixi.datastore.metadatastore/size-bytes
-                :kixi.datastore.metadatastore/file-type
-                :kixi.datastore.metadatastore/header
-                :kixi.datastore.metadatastore/name
-                :kixi.datastore.metadatastore/id
-                :kixi.datastore.metadatastore/type
-                :kixi.datastore.metadatastore/description
-                :kixi.datastore.metadatastore/provenance]))
+(s/def ::id (api-spec sc/uuid? "string"))
```
Comment (otfrom) [inline src/witan/httpapi/spec.clj:44]: I take it this is all internal spec and doesn't need to live in the shared spec library?

**Review by otfrom — CHANGES_REQUESTED** [review]
I think that eventually witan.httpapi.api will want to be split along the lines of the `(context` bits so that each resource type is in its own namespace. I'm happy that the code that is actually tied to the handlers is relatively small though. Just a function here that looks to have suffered a little bit that I think needs to be cleaned up.

File: src/witan/httpapi/spec.clj (hunk 4 above)
Comment (acron0) [reply src/witan/httpapi/spec.clj:44]: Correct

File: src/witan/httpapi/components/activities.clj (hunk 1 above)
Comment (jasebell) [reply src/witan/httpapi/components/activities.clj]: Response codes now defined in their own namespace see commit b572b01 for alterations.

**Review by otfrom — APPROVED** [review]

**Pull request merged**